### PR TITLE
fix: use QueryLogger for logging queries

### PR DESCRIPTION
**Pull request by vvcephei**
### Description 
We should always use the QueryLogger when logging query text, to allow anonymization in sensitive environments.

### Testing done 
Running CIT.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by vcrfxia** [comment]
Hey @vvcephei , this PR diverges from the existing logging approach in the codebase. There are two types of query masking/anonymization in the codebase today:
* QueryLogger anonymizes queries in such a way to only allow the structure of the query through. It's not nearly enough info to make for a useful log message, but is still useful for aggregating information about types of queries issued on a particular cluster.
* QueryMask for logging anonymization. This is a more-targeted anonymization mechanism to mask potentially sensitive info (configs in CREATE CONNECTOR statements, and values in INSERT VALUES statements) and the codebase is set up to ensure that whenever statement text is logged in log messages (including the line targeted in this PR), the text has passed through the QueryMask already. 

In other words, if you believe that the change in this PR is still needed (i.e., QueryMask doesn't provide a strong enough level of anonymization for the log statement you're targeting), then we'll want to make a bigger change (probably update QueryMask) so the analogous change gets propagated to all the other places we log statements. Or if you're happy with the current logging setup, then we can close the PR. 

(Here are the relevant PRs for context about the QueryMask setup: https://github.com/confluentinc/ksql/pull/9244, https://github.com/confluentinc/ksql/pull/9428)

**Comment by vvcephei** [comment]
Thanks for the clarification, @vcrfxia ! I'm happy now.
